packages/playwright-core/src/server/injected/recorder/recorder.ts

```diff
@@ -225,6 +225,10 @@ class RecordActionTool implements RecorderTool {
   }
 
   onClick(event: MouseEvent) {
+    // in webkit, when clicking a range input, event target is the body element and not the input itself.
```

Comment:
WebKit gives me <input> target, what do I do wrong?

<img width="282" alt="Screenshot 2024-01-30 at 11 42 06 AM" src="https://github.com/microsoft/playwright/assets/883973/93cb3a85-4c96-448e-b5a9-8d48c9b570d2">

Reply:
Apparently, it's my test `dragSlider` action that's causing the `body` click event.

Here is a test to check the target element of the click event:

```ts
  test('click range input', async ({ page }) => {
    await page.setContent(`<input type="range">`);
    await page.evaluate(() => {
      document.addEventListener('click', ({ target }) => console.log((target as HTMLElement).tagName), true);
    });
    const dragSlider = async () => {
      const { x, y, width, height } = await page.locator('input').boundingBox();
      await page.mouse.move(x + width / 2, y + height / 2);
      await page.mouse.down();
      await page.mouse.move(x + width, y + height / 2);
      await page.mouse.up();
    };

    const [msg] = await Promise.all([
      page.waitForEvent('console'),
      dragSlider(),
    ]);
    expect(msg.text()).toBe('INPUT');
  });
```

In chrome and firefox, the click event is triggered on the input element, in webkit it is on the body. I'm running those tests is windows, btw.

Reply:
The `click` event is triggered in the last hovered element, and because I'm moving to `x + width`, it's already in the border of the element and webkit already considers it the `body` element.

In my test above, it passes if mouse is released one pixel before, basically replacing:

```ts
await page.mouse.move(x + width, y + height / 2);
```

with:

```ts
await page.mouse.move(x + width - 1, y + height / 2);
```

But ir's normal that real users slide after the element width, so I think that the check is necessary.

I tried to record it manually without the ` isRangeInput(this._hoveredElement)` check, by running:

 ```
npx playwright codegen -b webkit 'data:text/html,<input type="range" min="0" max="10" value="5">' 
```

I get the following recorded lines:

```ts
  // if I restrain my mouse movements
  await page.getByRole('slider').fill('10');

  // if I move past the element (and the slider at the maximum value, that is, 10)
  await page.getByRole('slider').fill('5');
  await page.getByRole('slider').click();
```

Reply:
I see what you are saying. We are lucky that webkit does not dispatch the mouse moves as you drag the slider though. If it did, it would reset your accounting as hovered element would become body (or some other element). Also your comment is not entirely accurate - if you are clicking the range element, you are getting a click on input. It is only if you drag to outside you are getting a body target. Let's document this aspect in the comment.